packages/vitest/src/node/config.ts

```diff
@@ -122,13 +123,15 @@ export function resolveConfig(
   if (process.env.VITEST_MIN_THREADS)
     resolved.minThreads = parseInt(process.env.VITEST_MIN_THREADS)
 
-  resolved.setupFiles = toArray(resolved.setupFiles || []).map(file => resolve(resolved.root, file))
+  const require = createRequire(resolve(resolved.root, '_'))
```

Comment:
```suggestion
  const require = createRequire(resolve(resolved.root, '_'))
```

What's the `_` for?

Reply:
just a placeholder, `createRequire` needs a _file_, maybe using `index.js` more readable?

Reply:
Can we use `resolveModule` from `local-pkg`?

Reply:
I have tried, `resolveModule` uses `const _require = createRequire(import.meta.url)`, it's hard code as `import.meta.url`, but we need relative to `root` here. otherwise causes some bad cases.

Reply:
This will incorrectly resolve esm files. We use Vite's resolver, when importing files.

Reply:
https://github.com/vitest-dev/vitest/blob/03a54e0651f51b739b7833817d14d2cbcf474ede/packages/vitest/src/node/config.ts#L125

The `setupFiles` will prepend `config.root` path directly,
```js
{
  test: {
    setupFiles: ['@testing-library/jest-dom/extend-expect.js'],
  },
}
```

will be transformed to:
```js
{
  test: {
    setupFiles: ['/path/to/@testing-library/jest-dom/extend-expect.js'],
  },
}
```

it should be:
```js
{
  test: {
    setupFiles: ['/path/to/node_modules/@testing-library/jest-dom/extend-expect.js'],
  },
}
```